scapy/sessions.py

```diff
@@ -332,13 +336,16 @@ def _process_packet(self, pkt):
         # Stack the result on top of the previous frames
         if packet:
             data.clear()
-            metadata.clear()
             del self.tcp_frags[ident]
             pay.underlayer.remove_payload()
             if IP in pkt:
                 pkt[IP].len = None
                 pkt[IP].chksum = None
-            return pkt / packet
+            pkt_ret = pkt / packet
+            if self.store:
+                pkt_ret.pkt_list=metadata["pkt_list"]
```

Comment:
You gotta document this somewhere if you want this to be added. Adding a mysterious packet field isn't great

Reply:
The use case is when analysing [HTTP] packet flows using sniff(...`session=TCPSession`) which results in each request and response being extracted into 'session' packets (len=None). The proposed `pkt_list` maintains a `PacketList` of the actual packets that makes up each 'session' packet. This list of actual packets are useful for analysing the data flow rates at packet level granularity per response - this is useful when the responses can be quite large like media flows which can be constructed from thousands of actual packets. This is particularly useful now modern browsers pool TCP connections so different [media] type request/responses can be spread across different TCP sessions and the `pkt_list` allows for per media packet level analysis.

I wasn't sure what was the best approach as to choosing how to 'attach' the `pkt_list` to the session packets. I'd be happy to document it if people are happy with the approach and I'm open to suggestions as to better place to store the `pkt_list`. Also happy to fix any PEP8 issues.